src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,103 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    private static final int DEFAULT_CAPACITY = 10;
+    private T[] values;
+    private int size;
+
+    public ArrayList() {
+        values = (T[]) new Object[DEFAULT_CAPACITY];
+    }
+
     @Override
     public void add(T value) {
-
+        isNeadToResize();
+        values[size++] = value;
     }
 
     @Override
     public void add(T value, int index) {
-
+        isNeadToResize();
+        if (index != size) {
+            checkIndexWithSameSize(index);
```

Comment:
This check is better to move above resizing method, so that storage is not grown, if index is incorrect